crates/red_knot_python_semantic/src/types/call/arguments.rs

```diff
@@ -109,21 +109,6 @@ impl<'a, 'db> CallArgumentTypes<'a, 'db> {
         result
     }
 
-    /// Create a new [`CallArgumentTypes`] by prepending a synthetic argument to the front of this
-    /// argument list.
-    pub(crate) fn prepend_synthetic(&self, synthetic: Type<'db>) -> Self {
```

Comment:
This is redundant with `with_self`, as long as you can reuse the `CallArgumentTypes` instance